### PR TITLE
Refactor create deployment and add --port flag 

### PR DESCRIPTION
**Pull request by soltysh**
**What type of PR is this?**
/kind cleanup
/kind feature

**What this PR does / why we need it**:
This is coming from discussions from https://github.com/kubernetes/kubernetes/pull/87077 to add `--port` flag support for `kubectl create deployment` command.

**Special notes for your reviewer**:
/assign @sallyom @zhouya0 

**Does this PR introduce a user-facing change?**:
```release-note
Add --port flag to kubectl create deployment
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go [hunk 1]
```diff
@@ -33,123 +41,175 @@ var (
...
+	return &CreateDeploymentOptions{
+		Port:       -1,
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:  ioStreams,
+	}
 }
 
 // NewCmdCreateDeployment is a macro command to create a new deployment.
 // This command is better known to users as `kubectl create deployment`.
 // Note that this command overlaps significantly with the `kubectl run` command.
```
Comment (sftim) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go:81]: nit / aside: This line might be out of date now.

**Review by sftim — COMMENTED** [review]
Is there a test to validate the new command line option behaves as expected? (if there isn't, that might be OK - I'm asking for clarity so that the situation is clear to reviewers)

**Comment by soltysh** [comment]
/hold
I'll refactor this on top of https://github.com/kubernetes/kubernetes/pull/90676

**Comment by sftim** [comment]
Those look like flakes to me
/retest

**Comment by soltysh** [comment]
> Those look like flakes to me
> /retest

No worries, I was waiting for https://github.com/kubernetes/kubernetes/pull/91199 to land first :)

**Comment by zhouya0** [comment]
Seems testing fail with this:
https://github.com/kubernetes/kubernetes/blob/f097cee1566983cd1838f7d02b08ccc2b0921acd/test/cmd/apps.sh#L200

**Comment by soltysh** [comment]
Good catch, I missed that bit. I accidentally dropped the `--generator` flag which is deprecated but still has to be there.

**Pull request merged**